dkc/core/rest/file.py

```diff
@@ -86,7 +87,12 @@ def perform_create(self, serializer: FileSerializer):
         user: User = self.request.user
         if not folder.has_permission(user, permission=Permission.write):
             raise PermissionDenied()
-        serializer.save(creator=user)
+        try:
+            serializer.save(creator=user)
+        except QuotaLimitedError:
+            raise serializers.ValidationError(
+                {'size': ['This file would exceed the size quota for this folder.']}
```

Comment:
If there's an easier way to achieve this response structure, let me know. I want it to mirror the structure of the other types of validation failure responses.

Reply:
If the error occurs within a serializer field validator, the field will be automatically set. Since it's not thrown during validation, unfortunately it won't.

An alternative implementation could implement an internal API to determine whether a given allocation would exceed the quota. We'd then have to either do the entire API operation atomically, or we'd have to re-check the same thing on save anyway.

Since end users ultimately get the right response this way, and it seems simpler to implement (though slightly less "correct", since we want to try to do all validation in the serializer), I think the current way is best.

Reply:
Actually, I'll weaken my position: I think the current way is adequate. The alternative would require a new File/Folder/Quota API, but probably won't be any more complex on the DRF layer, since we'd no longer catch exceptions during `.save`.